Add a private copy of <linux/nfc.h>

The addition of the <linux/nfc.h> header in the kernel sources is
moderately recent, and therefore, a number of toolchains don't have
this header.

As a workaround until all reasonable toolchains get this header
included, add a configure check in neard to test whether this header
is available or not, and if not, use our own private copy of it.

Since symbols from the v3.7 kernel header are used, but linux/nfc.h
already exists since the v3.1 kernel, check for one of the most
recently added symbols: NFC_LLCP_RAW_HEADER_SIZE.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

Index: b/ncurses/base/MKlib_gen.sh
===================================================================
--- a/ncurses/base/MKlib_gen.sh
+++ b/ncurses/base/MKlib_gen.sh
@@ -62,7 +62,9 @@
 if test "${LC_CTYPE+set}"    = set; then LC_CTYPE=C;    export LC_CTYPE;    fi
 if test "${LC_COLLATE+set}"  = set; then LC_COLLATE=C;  export LC_COLLATE;  fi
 
-preprocessor="$1 -DNCURSES_INTERNALS -I../include"
+# Work around "unexpected" output of GCC 5.x's cpp w.r.t. #line directives
+# by simply suppressing them:
+preprocessor="$1 -P -DNCURSES_INTERNALS -I../include"
 AWK="$2"
 USE="$3"
 
